### PR TITLE
Revert hiding VTX tab

### PR DESCRIPTION
**Pull request by haslinghuis**
Inverting logic.

:thinking: Thinking of just removing the fake VTX feature in firmware (showing vtx tab permanent as before) as it's still seems to have issues. It's not a real feature as it does not show in CLI and rather want to reduce code.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by hydra** [comment]
From conversations yesterday:

1) the `buildOptions` are not present when devs build firmware, nor is there a way to create them (AFAIK).
2) in BF the firmware is the source of truth, so you must use that truth and anything else is/may-be a lie.
3) the FC does not need the VTX feature for anything, which is why it was deleted in https://github.com/cleanflight/cleanflight/pull/2721/files
4) the configurator could issue an MSP request for 'MSP_VTX_CONFIG' on connect to see if the FC responds, and this is 100% backwards compatible but slows down initial connect.
5) adding a 'capability' to MSP_BOARD_INFO is an option, but is not backwards compatible.

Given the above, the best solution is to always show the 'Video Transmitter' tab,  Like is already done for 'LED Strip', and then when the user clicks the 'Video Transmitter' tab the configurator issue the appropriate MSP requests and update the UI accordingly, like how the configurator updates the UI in say the 'Blackbox' tab via `supported` css styles, etc.  You also get to delete code in both the firmware and the configurator.  Also, since BF is mainly for FPV pilots it caters to the majority of the user-base, LoS pilots probably wouldn't care about seeing the 'Video Transmitter' tab.

**Review by nerdCopter — APPROVED** [review]

File: src/js/gui.js
Comment (hydra) [inline src/js/gui.js:62]: similarly, `defaultCloudBuildTabOptions` should be deleted for the same reasons as per my comment here: https://github.com/betaflight/betaflight-configurator/pull/3512#issuecomment-1635789348

Comment (haslinghuis) [reply src/js/gui.js:62]: Disagree. Without cloudbuild `defaultCloudBuildTabOptions` are concatenated into `defaultAllowedTabs.`

Comment (hydra) [reply src/js/gui.js:62]: 1/2) yes I understand that when not present the logic won't apply, but that doesn't address the fundamental issue in that the cloud build system is not the source of truth, as you can build firmware locally and connect to it using the configurator. The firmware should be used to check for feature availability in all cases, both cloud build and local build.

what we need is a solution that does not need any cloud build artifacts; Cloud build api use should be for building/flashing of firmware only, and not for configurator interaction with the FC firmware.

Comment (haslinghuis) [reply src/js/gui.js:62]: The idea behind this addition is to hide certain features with cloud build in mind.
Until now there are no issues reported except the virtual VTX feature which is addressed in the PR now.

> what we need is a solution that does not need any cloud build artifacts;

Yes would be nice to come up with a solution in firmware itself. Which extends to configuration of certain items like receiver etc.

Also we use cloud build options to preset options in a new build so user does not have to change again after flashing subsequent times.

**Comment by nerdCopter** [comment]
Although i also have concerns about buildOptions for gating, for now this PR should merge.

**Review by blckmn — APPROVED** [review]

**Pull request merged**
